onedal/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,215 @@
+# ==============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from abc import ABCMeta, abstractmethod
+from numbers import Number
+
+import numpy as np
+from sklearn.base import BaseEstimator
+
+from daal4py.sklearn._utils import get_dtype, make2d
+from onedal import _backend
+
+from ..common._estimator_checks import _check_is_fitted
+from ..common._mixin import ClassifierMixin
+from ..common._policy import _get_policy
+from ..datatypes import _convert_to_supported, from_table, to_table
+from ..utils import _check_array, _check_n_features, _check_X_y, _num_features, _type_of_target
+
+
+class BaseLogisticRegression(BaseEstimator, metaclass=ABCMeta):
+    @abstractmethod
+    def __init__(self, tol, C, fit_intercept, solver, max_iter, copy_X, algorithm):
+        self.tol = tol
+        self.C = C
+        self.fit_intercept = fit_intercept
+        self.solver = solver
+        self.max_iter = max_iter
+        self.algorithm = algorithm
+        self.copy_X = copy_X
+
+    def _get_policy(self, queue, *data):
+        return _get_policy(queue, *data)
+
+    def _get_onedal_params(self, dtype=np.float32):
+        intercept = "intercept|" if self.fit_intercept else ""
+        return {
+            "fptype": "float" if dtype == np.float32 else "double",
+            "method": self.algorithm,
+            "intercept": self.fit_intercept,
+            "tol": self.tol,
+            "max_iter": self.max_iter,
+            "C" : self.C,
+            "optimizer": self.solver,
+            "result_option": (intercept + "coefficients|iterations_count"),
+        }
+
+    def _fit(self, X, y, module, queue):
+        policy = self._get_policy(queue, X, y)
+
+        X_loc, y_loc = X, y
+        if not isinstance(X, np.ndarray):
```

Comment:
numpy.asarray already checks for numpy ndarray instance, no need for that. Please remove